production/helm/loki/templates/_helpers.tpl

```diff
@@ -238,46 +264,55 @@ s3:
   {{- with .signatureVersion }}
   signature_version: {{ . }}
   {{- end }}
-  s3forcepathstyle: {{ .s3ForcePathStyle }}
-  insecure: {{ .insecure }}
   {{- with .disable_dualstack }}
   disable_dualstack: {{ . }}
   {{- end }}
-  {{- with .http_config}}
+  {{- with .http_config }}
   http_config:
-{{ toYaml . | indent 4 }}
+    {{- toYaml . | nindent 4 }}
   {{- end }}
-  {{- with .backoff_config}}
+  {{- with .backoff_config }}
   backoff_config:
-{{ toYaml . | indent 4 }}
+    {{- toYaml . | nindent 4 }}
   {{- end }}
   {{- with .sse }}
   sse:
-{{ toYaml . | indent 4 }}
+    {{- toYaml . | nindent 4 }}
   {{- end }}
-{{- end -}}
-
-{{- else if eq .Values.loki.storage.type "gcs" -}}
-{{- with .Values.loki.storage.gcs }}
+{{- end }}
+{{- else if eq .ctx.Values.loki.storage.type "gcs" -}}
+{{- with .ctx.Values.loki.storage.gcs }}
 gcs:
-  bucket_name: {{ $.Values.loki.storage.bucketNames.chunks }}
-  chunk_buffer_size: {{ .chunkBufferSize }}
-  request_timeout: {{ .requestTimeout }}
-  enable_http2: {{ .enableHttp2 }}
-{{- end -}}
-{{- else if eq .Values.loki.storage.type "azure" -}}
-{{- with .Values.loki.storage.azure }}
+  {{- toYaml (mergeOverwrite
+    dict
+    (dict
+      "bucket_name" $bucketName
+      "chunk_buffer_size" .chunkBufferSize
+      "request_timeout" .requestTimeout
+      "enable_http2" .enableHttp2
+    )
+    (omit . "bucket_name" "chunkBufferSize" "requestTimeout" "enableHttp2")
+  ) | nindent 2 }}
+{{- end -}}
+{{- else if eq .ctx.Values.loki.storage.type "azure" -}}
+{{- with .ctx.Values.loki.storage.azure }}
 azure:
-  account_name: {{ .accountName }}
+  {{- toYaml (mergeOverwrite
+    dict
```

Comment:
this is not easily readable 🤔 Can we do it a bit differently? Maybe azure, gcs and s3 could be extracted into separate templates for better readibility?

Reply:
please check this out if it meets your expectations